python/pyspark/sql/connect/dataframe.py

```diff
@@ -875,6 +877,60 @@ def to_jcols(
 
     melt = unpivot
 
+    def randomSplit(
+        self,
+        weights: List[float],
+        seed: Optional[int] = None,
+    ) -> List["DataFrame"]:
+        """Randomly splits this :class:`DataFrame` with the provided weights.
+
+        .. versionadded:: 3.4.0
+
+        Parameters
+        ----------
+        weights : list
+            list of doubles as weights with which to split the :class:`DataFrame`.
+            Weights will be normalized if they don't sum up to 1.0.
+        seed : int, optional
+            The seed for sampling.
+
+        Returns
+        -------
+        list
+            List of DataFrames.
+        """
+        for w in weights:
+            if w < 0.0:
+                raise ValueError("Weights must be positive. Found weight value: %s" % w)
```

Comment:
this checking is similar to the one in PySpark dataframe.py, but for the below checks we should move them to the server make sure that all clients benefit from the logic.

Reply:
More lightweight to check at client.